pkg/pool-manager/pool.go

```diff
@@ -26,26 +26,34 @@ import (
 )
 
 const (
-	StartPoolRangeEnv  = "START_POOL_RANGE"
-	EndPoolRangeEnv    = "END_POOL_RANGE"
-	networksAnnotation = "k8s.v1.cni.cncf.io/networks"
+	StartPoolRangeEnv        = "START_POOL_RANGE"
+	EndPoolRangeEnv          = "END_POOL_RANGE"
+	networksAnnotation       = "k8s.v1.cni.cncf.io/networks"
+	networksStatusAnnotation = "k8s.v1.cni.cncf.io/networks-status"
 )
 
 var log = logf.Log.WithName("PoolManager")
 
 type PoolManager struct {
-	kubeClient      kubernetes.Interface // kubernetes client
-	startRange      net.HardwareAddr     // fist mac in range
-	endRange        net.HardwareAddr     // last mac in range
-	currentMac      net.HardwareAddr     // last given mac
-	macPoolMap      map[string]bool      // allocated mac map
-	podToMacPoolMap map[string][]string  // map for namespace/podname and a list of allocated mac addresses
-	vmToMacPoolMap  map[string][]string  // cap for namespace/vmname and a list of allocated mac addresses
-	poolMutex       sync.Mutex           // mutex for allocation an release
-	isLeader        bool                 // leader boolean
-	isKubevirt      bool                 // bool if kubevirt virtualmachine crd exist in the cluster
+	kubeClient      kubernetes.Interface         // kubernetes client
+	startRange      net.HardwareAddr             // fist mac in range
+	endRange        net.HardwareAddr             // last mac in range
+	currentMac      net.HardwareAddr             // last given mac
+	macPoolMap      map[string]AllocationStatus  // allocated mac map and status
+	podToMacPoolMap map[string]map[string]string // map allocated mac address by networkname and namespace/podname: {"namespace/podname: {"network name": "mac address"}}
+	vmToMacPoolMap  map[string][]string          // cap for namespace/vmname and a list of allocated mac addresses
+	poolMutex       sync.Mutex                   // mutex for allocation an release
+	isLeader        bool                         // leader boolean
+	isKubevirt      bool                         // bool if kubevirt virtualmachine crd exist in the cluster
 }
 
+type AllocationStatus string
+
+const (
+	AllocationStatusAllocated     AllocationStatus = "Allocated"
+	AllocationStatusWaitingForPod AllocationStatus = "WaitingForPod"
```

Comment:
I see it being assigned, but never changed to "Allocated". Shouldn't we set it to "Allocated" somewhere?

Reply:
https://github.com/K8sNetworkPlumbingWG/kubemacpool/pull/9/files/c219fd40a10bbfb5c0c0810e721e3e1981a8cc6b..789c062254a1fccc8b8e6211e12b642f26eab7e3#diff-f178743f06898ca1ae8f344f45065660R179

here is the change

Reply:
I see, all good then